shiny_app/server.R

```diff
@@ -351,4 +351,153 @@ shinyServer(function(input, output, session) {
+      dplyr::group_by(condition) %>%
+      filter(!is.na(condition)) %>%
+      summarise(`Adjusted exposed / at risk @ baseline` = as.integer(round(sum(`new-at-risk`, na.rm = T), 0)),
+                `Total symptomatic over period referrals` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+                `Total receiving services over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+      arrange(-`Total symptomatic over period referrals`)
+  })
+
+
+  ## Tab - Treatment Pathway ####
+
+  summary_treatment_pathway <- reactive({
+    summary_outputs() %>%
+    pivot_wider(names_from = type, values_from = value) %>%
+    dplyr::group_by(treatment_pathway) %>%
+    filter(!is.na(treatment_pathway)) %>%
+    summarise(`Total new referrals/presentations over period` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+              `Total services offered over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+    arrange(-`Total new referrals/presentations over period`)
+  })
+
+  output$surge_treatmentpathway <- renderTable({
+    summary_treatment_pathway()
+  })
+
+  output$surge_treatmentpathwayplot <- renderPlot({
+    summary_treatment_pathway() %>%
+      pivot_longer(!contains("treatment_pathway")) %>%
+      ggplot(aes(treatment_pathway, value)) +
+      geom_col(position = "dodge") +
+      labs(y = "Number",
+             x = "Treatment Pathway") +
+      scale_x_discrete(labels = function(x) str_wrap(x, 8),
+                       guide = guide_axis(n.dodge = 2)) +
+      facet_wrap(~ name, ncol=1)
+
+  })
+
+  ## Bubble Pack testing ####
+
+  output$bubble_plot_baselinepopn <- renderPlotly({
+
+    circle_pack_plot <- map_dbl(params$groups, "size") %>%
+      enframe(name = "subpopn") %>%
+      mutate(level_2 = case_when(subpopn %in% c("Children & young people", "Students FE & HE") ~ "Children & young people",
+                                 subpopn %in% c("Elderly alone") ~ "Elderly alone",
+                                 subpopn %in% c("General population") ~ "General population",
+                                 subpopn %in% c("Domestic abuse victims", "Family of COVID deceased", "Family of ICU survivors", "Newly unemployed", "Pregnant & New Mothers", "Parents") ~ "Other Adults and Specific Groups",
+                                 subpopn %in% c("Health and care workers", "ICU survivors") ~ "Directly affected individuals",
+                                 subpopn %in% c("Learning disabilities & autism", "Pre existing CMH illness", "Pre existing LTC", "Pre existing SMI") ~ "Existing Conditions",
+                                 TRUE ~ NA_character_))
+
+
+
+    packing <- circleProgressiveLayout(circle_pack_plot$value, sizetype='area')
+    circle_pack_plot <- cbind(circle_pack_plot, packing)
+    dat.gg <- circleLayoutVertices(packing, npoints=50)
+
+    dat.gg <- dat.gg %>% left_join(tibble(level_2 = circle_pack_plot$level_2, id = 1:16),
```

Comment:
merge lines 439 and 441 into a single pipeline

Reply:
merged lines